app/ui/lib/Modal.tsx

```diff
@@ -125,6 +126,21 @@ Modal.Body = classed.div`py-2 overflow-y-auto`
 
 Modal.Section = classed.div`p-4 space-y-4 border-b border-secondary text-secondary last-of-type:border-none text-sans-md`
 
+/**
+ * `formId` and `onAction` are mutually exclusive. If there is a form associated,
+ * the button becomes a submit button for that form, and the action is assumed to
+ * be hooked up in the form's `onSubmit`.
+ */
+type FooterProps = {
+  children?: React.ReactNode
+  onDismiss: () => void
+  actionType?: 'primary' | 'danger'
+  actionText: React.ReactNode
+  actionLoading?: boolean
+  cancelText?: string
+  disabled?: boolean
+} & MergeExclusive<{ formId: string }, { onAction: () => void }>
+
```

Comment:
yoooo `MergeExclusive` is my new best friend

Reply:
🤯